small/matras.c

```diff
@@ -364,6 +366,27 @@ matras_touch_reserve(struct matras *m, int count)
 	return matras_allocator_reserve(m->allocator, max_extents_required);
 }
 
+/**
+ * Reserve the max amount of extents required to successfully allocate @p count
+ * blocks. The extents are reserved in the allocator given on construction.
+ */
+int
+matras_alloc_reserve(struct matras *m, int count)
+{
+	assert(count >= 0);
+
+	/* No allocations planned. */
+	if (count == 0)
+		return 0;
+
+	/* The numbers are higher than possibly actual amount. */
+	int l3_count = DIV_ROUND_UP(m->block_size * count,
+				    m->allocator->extent_size);
+	int l2_count = DIV_ROUND_UP(l3_count * sizeof(void *),
+				    m->allocator->extent_size);
+	return matras_allocator_reserve(m->allocator, l3_count + l2_count + 1);
```

Comment:
Shouldn't you take into account already allocated blocks?

Reply:
We can, the current approach is just simpler and gives up to 48kb overhead, which is negligible, since the reserved memory is used in each memtx index.